// zohocrm/src/model/Account.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Zohocrm) {
      root.Zohocrm = {};
    }
    root.Zohocrm.Account = factory(root.Zohocrm.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The Account model module.
   * @module model/Account
   * @version api-v2
   */

  /**
   * Constructs a new <code>Account</code>.
   * @alias module:model/Account
   * @class
   * @param accountName
   */
  var exports = function(accountName) {


    this['Account Name'] = accountName;



































  };

  /**
   * Constructs a <code>Account</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Account} obj Optional instance to populate.
   * @return {module:model/Account} The populated <code>Account</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('ACCOUNTID')) {
        obj['ACCOUNTID'] = ApiClient.convertToType(data['ACCOUNTID'], 'String');
      }
      if (data.hasOwnProperty('Account Name')) {
        obj['Account Name'] = ApiClient.convertToType(data['Account Name'], 'String');
      }
      if (data.hasOwnProperty('Account Number')) {
        obj['Account Number'] = ApiClient.convertToType(data['Account Number'], 'String');
      }
      if (data.hasOwnProperty('Account Owner')) {
        obj['Account Owner'] = ApiClient.convertToType(data['Account Owner'], 'String');
      }
      if (data.hasOwnProperty('Account Site')) {
        obj['Account Site'] = ApiClient.convertToType(data['Account Site'], 'String');
      }
      if (data.hasOwnProperty('Account Type')) {
        obj['Account Type'] = ApiClient.convertToType(data['Account Type'], 'String');
      }
      if (data.hasOwnProperty('Annual Revenue')) {
        obj['Annual Revenue'] = ApiClient.convertToType(data['Annual Revenue'], 'String');
      }
      if (data.hasOwnProperty('Billing City')) {
        obj['Billing City'] = ApiClient.convertToType(data['Billing City'], 'String');
      }
      if (data.hasOwnProperty('Billing Code')) {
        obj['Billing Code'] = ApiClient.convertToType(data['Billing Code'], 'String');
      }
      if (data.hasOwnProperty('Billing Country')) {
        obj['Billing Country'] = ApiClient.convertToType(data['Billing Country'], 'String');
      }
      if (data.hasOwnProperty('Billing State')) {
        obj['Billing State'] = ApiClient.convertToType(data['Billing State'], 'String');
      }
      if (data.hasOwnProperty('Billing Street')) {
        obj['Billing Street'] = ApiClient.convertToType(data['Billing Street'], 'String');
      }
      if (data.hasOwnProperty('Created By')) {
        obj['Created By'] = ApiClient.convertToType(data['Created By'], 'String');
      }
      if (data.hasOwnProperty('Created Time')) {
        obj['Created Time'] = ApiClient.convertToType(data['Created Time'], 'String');
      }
      if (data.hasOwnProperty('Description')) {
        obj['Description'] = ApiClient.convertToType(data['Description'], 'String');
      }
      if (data.hasOwnProperty('Employees')) {
        obj['Employees'] = ApiClient.convertToType(data['Employees'], 'String');
      }
      if (data.hasOwnProperty('Fax')) {
        obj['Fax'] = ApiClient.convertToType(data['Fax'], 'String');
      }
      if (data.hasOwnProperty('Industry')) {
        obj['Industry'] = ApiClient.convertToType(data['Industry'], 'String');
      }
      if (data.hasOwnProperty('Last Activity Time')) {
        obj['Last Activity Time'] = ApiClient.convertToType(data['Last Activity Time'], 'String');
      }
      if (data.hasOwnProperty('MODIFIEDBY')) {
        obj['MODIFIEDBY'] = ApiClient.convertToType(data['MODIFIEDBY'], 'String');
      }
      if (data.hasOwnProperty('Modified By')) {
        obj['Modified By'] = ApiClient.convertToType(data['Modified By'], 'String');
      }
      if (data.hasOwnProperty('Modified Time')) {
        obj['Modified Time'] = ApiClient.convertToType(data['Modified Time'], 'String');
      }
      if (data.hasOwnProperty('Ownership')) {
        obj['Ownership'] = ApiClient.convertToType(data['Ownership'], 'String');
      }
      if (data.hasOwnProperty('PARENTACCOUNTID')) {
        obj['PARENTACCOUNTID'] = ApiClient.convertToType(data['PARENTACCOUNTID'], 'String');
      }
      if (data.hasOwnProperty('Parent Account')) {
        obj['Parent Account'] = ApiClient.convertToType(data['Parent Account'], 'String');
      }
      if (data.hasOwnProperty('Phone')) {
        obj['Phone'] = ApiClient.convertToType(data['Phone'], 'String');
      }
      if (data.hasOwnProperty('Rating')) {
        obj['Rating'] = ApiClient.convertToType(data['Rating'], 'String');
      }
      if (data.hasOwnProperty('SIC Code')) {
        obj['SIC Code'] = ApiClient.convertToType(data['SIC Code'], 'String');
      }
      if (data.hasOwnProperty('SMCREATORID')) {
        obj['SMCREATORID'] = ApiClient.convertToType(data['SMCREATORID'], 'String');
      }
      if (data.hasOwnProperty('SMOWNERID')) {
        obj['SMOWNERID'] = ApiClient.convertToType(data['SMOWNERID'], 'String');
      }
      if (data.hasOwnProperty('Shipping City')) {
        obj['Shipping City'] = ApiClient.convertToType(data['Shipping City'], 'String');
      }
      if (data.hasOwnProperty('Shipping Code')) {
        obj['Shipping Code'] = ApiClient.convertToType(data['Shipping Code'], 'String');
      }
      if (data.hasOwnProperty('Shipping Country')) {
        obj['Shipping Country'] = ApiClient.convertToType(data['Shipping Country'], 'String');
      }
      if (data.hasOwnProperty('Shipping State')) {
        obj['Shipping State'] = ApiClient.convertToType(data['Shipping State'], 'String');
      }
      if (data.hasOwnProperty('Shipping Street')) {
        obj['Shipping Street'] = ApiClient.convertToType(data['Shipping Street'], 'String');
      }
      if (data.hasOwnProperty('Ticker Symbol')) {
        obj['Ticker Symbol'] = ApiClient.convertToType(data['Ticker Symbol'], 'String');
      }
      if (data.hasOwnProperty('Website')) {
        obj['Website'] = ApiClient.convertToType(data['Website'], 'String');
      }
    }
    return obj;
  }


  /**
   * @member {String} ACCOUNTID
   */
  exports.prototype['ACCOUNTID'] = undefined;

  /**
   * @member {String} Account Name
   */
  exports.prototype['Account Name'] = undefined;

  /**
   * @member {String} Account Number
   */
  exports.prototype['Account Number'] = undefined;

  /**
   * @member {String} Account Owner
   */
  exports.prototype['Account Owner'] = undefined;

  /**
   * @member {String} Account Site
   */
  exports.prototype['Account Site'] = undefined;

  /**
   * @member {String} Account Type
   */
  exports.prototype['Account Type'] = undefined;

  /**
   * @member {String} Annual Revenue
   */
  exports.prototype['Annual Revenue'] = undefined;

  /**
   * @member {String} Billing City
   */
  exports.prototype['Billing City'] = undefined;

  /**
   * @member {String} Billing Code
   */
  exports.prototype['Billing Code'] = undefined;

  /**
   * @member {String} Billing Country
   */
  exports.prototype['Billing Country'] = undefined;

  /**
   * @member {String} Billing State
   */
  exports.prototype['Billing State'] = undefined;

  /**
   * @member {String} Billing Street
   */
  exports.prototype['Billing Street'] = undefined;

  /**
   * @member {String} Created By
   */
  exports.prototype['Created By'] = undefined;

  /**
   * @member {String} Created Time
   */
  exports.prototype['Created Time'] = undefined;

  /**
   * @member {String} Description
   */
  exports.prototype['Description'] = undefined;

  /**
   * @member {String} Employees
   */
  exports.prototype['Employees'] = undefined;

  /**
   * @member {String} Fax
   */
  exports.prototype['Fax'] = undefined;

  /**
   * @member {String} Industry
   */
  exports.prototype['Industry'] = undefined;

  /**
   * @member {String} Last Activity Time
   */
  exports.prototype['Last Activity Time'] = undefined;

  /**
   * @member {String} MODIFIEDBY
   */
  exports.prototype['MODIFIEDBY'] = undefined;

  /**
   * @member {String} Modified By
   */
  exports.prototype['Modified By'] = undefined;

  /**
   * @member {String} Modified Time
   */
  exports.prototype['Modified Time'] = undefined;

  /**
   * @member {String} Ownership
   */
  exports.prototype['Ownership'] = undefined;

  /**
   * @member {String} PARENTACCOUNTID
   */
  exports.prototype['PARENTACCOUNTID'] = undefined;

  /**
   * @member {String} Parent Account
   */
  exports.prototype['Parent Account'] = undefined;

  /**
   * @member {String} Phone
   */
  exports.prototype['Phone'] = undefined;

  /**
   * @member {String} Rating
   */
  exports.prototype['Rating'] = undefined;

  /**
   * @member {String} SIC Code
   */
  exports.prototype['SIC Code'] = undefined;

  /**
   * @member {String} SMCREATORID
   */
  exports.prototype['SMCREATORID'] = undefined;

  /**
   * @member {String} SMOWNERID
   */
  exports.prototype['SMOWNERID'] = undefined;

  /**
   * @member {String} Shipping City
   */
  exports.prototype['Shipping City'] = undefined;

  /**
   * @member {String} Shipping Code
   */
  exports.prototype['Shipping Code'] = undefined;

  /**
   * @member {String} Shipping Country
   */
  exports.prototype['Shipping Country'] = undefined;

  /**
   * @member {String} Shipping State
   */
  exports.prototype['Shipping State'] = undefined;

  /**
   * @member {String} Shipping Street
   */
  exports.prototype['Shipping Street'] = undefined;

  /**
   * @member {String} Ticker Symbol
   */
  exports.prototype['Ticker Symbol'] = undefined;

  /**
   * @member {String} Website
   */
  exports.prototype['Website'] = undefined;




  return exports;
}));
