// zohocrm/src/model/BulkStatus.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Zohocrm) {
      root.Zohocrm = {};
    }
    root.Zohocrm.BulkStatus = factory(root.Zohocrm.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The BulkStatus model module.
   * @module model/BulkStatus
   * @version api-v2
   */

  /**
   * Constructs a new <code>BulkStatus</code>.
   * Status of the bulk upload.
   * @alias module:model/BulkStatus
   * @class
   */
  var exports = function() {







  };

  /**
   * Constructs a <code>BulkStatus</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/BulkStatus} obj Optional instance to populate.
   * @return {module:model/BulkStatus} The populated <code>BulkStatus</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('batchId')) {
        obj['batchId'] = ApiClient.convertToType(data['batchId'], 'Number');
      }
      if (data.hasOwnProperty('message')) {
        obj['message'] = ApiClient.convertToType(data['message'], 'String');
      }
      if (data.hasOwnProperty('numOfLeadsProcessed')) {
        obj['numOfLeadsProcessed'] = ApiClient.convertToType(data['numOfLeadsProcessed'], 'Number');
      }
      if (data.hasOwnProperty('numOfRowsFailed')) {
        obj['numOfRowsFailed'] = ApiClient.convertToType(data['numOfRowsFailed'], 'Number');
      }
      if (data.hasOwnProperty('numOfRowsWithWarning')) {
        obj['numOfRowsWithWarning'] = ApiClient.convertToType(data['numOfRowsWithWarning'], 'Number');
      }
      if (data.hasOwnProperty('status')) {
        obj['status'] = ApiClient.convertToType(data['status'], 'String');
      }
    }
    return obj;
  }


  /**
   * Contact/Leads loaded
   * @member {Number} batchId
   */
  exports.prototype['batchId'] = undefined;

  /**
   * @member {String} message
   */
  exports.prototype['message'] = undefined;

  /**
   * Contact/Leads loaded
   * @member {Number} numOfLeadsProcessed
   */
  exports.prototype['numOfLeadsProcessed'] = undefined;

  /**
   * Failed count
   * @member {Number} numOfRowsFailed
   */
  exports.prototype['numOfRowsFailed'] = undefined;

  /**
   * Contact/Leads that had warnings
   * @member {Number} numOfRowsWithWarning
   */
  exports.prototype['numOfRowsWithWarning'] = undefined;

  /**
   * Status of the bulk job
   * @member {String} status
   */
  exports.prototype['status'] = undefined;




  return exports;
}));
