// zohocrm/src/model/Calls.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Zohocrm) {
      root.Zohocrm = {};
    }
    root.Zohocrm.Calls = factory(root.Zohocrm.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The Calls model module.
   * @module model/Calls
   * @version api-v2
   */

  /**
   * Constructs a new <code>Calls</code>.
   * @alias module:model/Calls
   * @class
   */
  var exports = function() {























  };

  /**
   * Constructs a <code>Calls</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Calls} obj Optional instance to populate.
   * @return {module:model/Calls} The populated <code>Calls</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('ACTIVITYID')) {
        obj['ACTIVITYID'] = ApiClient.convertToType(data['ACTIVITYID'], 'String');
      }
      if (data.hasOwnProperty('Billable')) {
        obj['Billable'] = ApiClient.convertToType(data['Billable'], 'Boolean');
      }
      if (data.hasOwnProperty('CONTACTID')) {
        obj['CONTACTID'] = ApiClient.convertToType(data['CONTACTID'], 'String');
      }
      if (data.hasOwnProperty('Call Duration')) {
        obj['Call Duration'] = ApiClient.convertToType(data['Call Duration'], 'String');
      }
      if (data.hasOwnProperty('Call Duration (in seconds)')) {
        obj['Call Duration (in seconds)'] = ApiClient.convertToType(data['Call Duration (in seconds)'], 'String');
      }
      if (data.hasOwnProperty('Call Owner')) {
        obj['Call Owner'] = ApiClient.convertToType(data['Call Owner'], 'String');
      }
      if (data.hasOwnProperty('Call Purpose')) {
        obj['Call Purpose'] = ApiClient.convertToType(data['Call Purpose'], 'String');
      }
      if (data.hasOwnProperty('Call Status')) {
        obj['Call Status'] = ApiClient.convertToType(data['Call Status'], 'String');
      }
      if (data.hasOwnProperty('Call Type')) {
        obj['Call Type'] = ApiClient.convertToType(data['Call Type'], 'String');
      }
      if (data.hasOwnProperty('Contact Name')) {
        obj['Contact Name'] = ApiClient.convertToType(data['Contact Name'], 'String');
      }
      if (data.hasOwnProperty('Created By')) {
        obj['Created By'] = ApiClient.convertToType(data['Created By'], 'String');
      }
      if (data.hasOwnProperty('Created Time')) {
        obj['Created Time'] = ApiClient.convertToType(data['Created Time'], 'Date');
      }
      if (data.hasOwnProperty('Description')) {
        obj['Description'] = ApiClient.convertToType(data['Description'], 'String');
      }
      if (data.hasOwnProperty('MODIFIEDBY')) {
        obj['MODIFIEDBY'] = ApiClient.convertToType(data['MODIFIEDBY'], 'String');
      }
      if (data.hasOwnProperty('Modified By')) {
        obj['Modified By'] = ApiClient.convertToType(data['Modified By'], 'String');
      }
      if (data.hasOwnProperty('Modified Time')) {
        obj['Modified Time'] = ApiClient.convertToType(data['Modified Time'], 'Date');
      }
      if (data.hasOwnProperty('RELATEDTOID')) {
        obj['RELATEDTOID'] = ApiClient.convertToType(data['RELATEDTOID'], 'String');
      }
      if (data.hasOwnProperty('Related To')) {
        obj['Related To'] = ApiClient.convertToType(data['Related To'], 'String');
      }
      if (data.hasOwnProperty('SEMODULE')) {
        obj['SEMODULE'] = ApiClient.convertToType(data['SEMODULE'], 'String');
      }
      if (data.hasOwnProperty('SMCREATORID')) {
        obj['SMCREATORID'] = ApiClient.convertToType(data['SMCREATORID'], 'String');
      }
      if (data.hasOwnProperty('SMOWNERID')) {
        obj['SMOWNERID'] = ApiClient.convertToType(data['SMOWNERID'], 'String');
      }
      if (data.hasOwnProperty('Subject')) {
        obj['Subject'] = ApiClient.convertToType(data['Subject'], 'String');
      }
    }
    return obj;
  }


  /**
   * @member {String} ACTIVITYID
   */
  exports.prototype['ACTIVITYID'] = undefined;

  /**
   * @member {Boolean} Billable
   */
  exports.prototype['Billable'] = undefined;

  /**
   * @member {String} CONTACTID
   */
  exports.prototype['CONTACTID'] = undefined;

  /**
   * @member {String} Call Duration
   */
  exports.prototype['Call Duration'] = undefined;

  /**
   * @member {String} Call Duration (in seconds)
   */
  exports.prototype['Call Duration (in seconds)'] = undefined;

  /**
   * @member {String} Call Owner
   */
  exports.prototype['Call Owner'] = undefined;

  /**
   * @member {String} Call Purpose
   */
  exports.prototype['Call Purpose'] = undefined;

  /**
   * @member {String} Call Status
   */
  exports.prototype['Call Status'] = undefined;

  /**
   * @member {String} Call Type
   */
  exports.prototype['Call Type'] = undefined;

  /**
   * @member {String} Contact Name
   */
  exports.prototype['Contact Name'] = undefined;

  /**
   * @member {String} Created By
   */
  exports.prototype['Created By'] = undefined;

  /**
   * @member {Date} Created Time
   */
  exports.prototype['Created Time'] = undefined;

  /**
   * @member {String} Description
   */
  exports.prototype['Description'] = undefined;

  /**
   * @member {String} MODIFIEDBY
   */
  exports.prototype['MODIFIEDBY'] = undefined;

  /**
   * @member {String} Modified By
   */
  exports.prototype['Modified By'] = undefined;

  /**
   * @member {Date} Modified Time
   */
  exports.prototype['Modified Time'] = undefined;

  /**
   * @member {String} RELATEDTOID
   */
  exports.prototype['RELATEDTOID'] = undefined;

  /**
   * @member {String} Related To
   */
  exports.prototype['Related To'] = undefined;

  /**
   * @member {String} SEMODULE
   */
  exports.prototype['SEMODULE'] = undefined;

  /**
   * @member {String} SMCREATORID
   */
  exports.prototype['SMCREATORID'] = undefined;

  /**
   * @member {String} SMOWNERID
   */
  exports.prototype['SMOWNERID'] = undefined;

  /**
   * @member {String} Subject
   */
  exports.prototype['Subject'] = undefined;




  return exports;
}));
