// zohocrm/src/model/Campaign.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Zohocrm) {
      root.Zohocrm = {};
    }
    root.Zohocrm.Campaign = factory(root.Zohocrm.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The Campaign model module.
   * @module model/Campaign
   * @version api-v2
   */

  /**
   * Constructs a new <code>Campaign</code>.
   * @alias module:model/Campaign
   * @class
   */
  var exports = function() {




















  };

  /**
   * Constructs a <code>Campaign</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Campaign} obj Optional instance to populate.
   * @return {module:model/Campaign} The populated <code>Campaign</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('Status')) {
        obj['Status'] = ApiClient.convertToType(data['Status'], 'String');
      }
      if (data.hasOwnProperty('Description')) {
        obj['Description'] = ApiClient.convertToType(data['Description'], 'String');
      }
      if (data.hasOwnProperty('MODIFIEDBY')) {
        obj['MODIFIEDBY'] = ApiClient.convertToType(data['MODIFIEDBY'], 'String');
      }
      if (data.hasOwnProperty('SMCREATORID')) {
        obj['SMCREATORID'] = ApiClient.convertToType(data['SMCREATORID'], 'String');
      }
      if (data.hasOwnProperty('SMOWNERID')) {
        obj['SMOWNERID'] = ApiClient.convertToType(data['SMOWNERID'], 'String');
      }
      if (data.hasOwnProperty('Expected Revenue')) {
        obj['Expected Revenue'] = ApiClient.convertToType(data['Expected Revenue'], 'String');
      }
      if (data.hasOwnProperty('Modified Time')) {
        obj['Modified Time'] = ApiClient.convertToType(data['Modified Time'], 'String');
      }
      if (data.hasOwnProperty('Created Time')) {
        obj['Created Time'] = ApiClient.convertToType(data['Created Time'], 'String');
      }
      if (data.hasOwnProperty('Modified By')) {
        obj['Modified By'] = ApiClient.convertToType(data['Modified By'], 'String');
      }
      if (data.hasOwnProperty('Actual Cost')) {
        obj['Actual Cost'] = ApiClient.convertToType(data['Actual Cost'], 'String');
      }
      if (data.hasOwnProperty('Type')) {
        obj['Type'] = ApiClient.convertToType(data['Type'], 'String');
      }
      if (data.hasOwnProperty('Start Date')) {
        obj['Start Date'] = ApiClient.convertToType(data['Start Date'], 'String');
      }
      if (data.hasOwnProperty('Expected Response')) {
        obj['Expected Response'] = ApiClient.convertToType(data['Expected Response'], 'String');
      }
      if (data.hasOwnProperty('Budgeted Cost')) {
        obj['Budgeted Cost'] = ApiClient.convertToType(data['Budgeted Cost'], 'String');
      }
      if (data.hasOwnProperty('Campaign Owner')) {
        obj['Campaign Owner'] = ApiClient.convertToType(data['Campaign Owner'], 'String');
      }
      if (data.hasOwnProperty('Created By')) {
        obj['Created By'] = ApiClient.convertToType(data['Created By'], 'String');
      }
      if (data.hasOwnProperty('CAMPAIGNID')) {
        obj['CAMPAIGNID'] = ApiClient.convertToType(data['CAMPAIGNID'], 'String');
      }
      if (data.hasOwnProperty('Campaign Name')) {
        obj['Campaign Name'] = ApiClient.convertToType(data['Campaign Name'], 'String');
      }
      if (data.hasOwnProperty('Num sent')) {
        obj['Num sent'] = ApiClient.convertToType(data['Num sent'], 'String');
      }
    }
    return obj;
  }


  /**
   * Valid values are 'Planning','Active','InActive', 'Complete'
   * @member {String} Status
   */
  exports.prototype['Status'] = undefined;

  /**
   * @member {String} Description
   */
  exports.prototype['Description'] = undefined;

  /**
   * @member {String} MODIFIEDBY
   */
  exports.prototype['MODIFIEDBY'] = undefined;

  /**
   * @member {String} SMCREATORID
   */
  exports.prototype['SMCREATORID'] = undefined;

  /**
   * @member {String} SMOWNERID
   */
  exports.prototype['SMOWNERID'] = undefined;

  /**
   * @member {String} Expected Revenue
   */
  exports.prototype['Expected Revenue'] = undefined;

  /**
   * @member {String} Modified Time
   */
  exports.prototype['Modified Time'] = undefined;

  /**
   * @member {String} Created Time
   */
  exports.prototype['Created Time'] = undefined;

  /**
   * @member {String} Modified By
   */
  exports.prototype['Modified By'] = undefined;

  /**
   * @member {String} Actual Cost
   */
  exports.prototype['Actual Cost'] = undefined;

  /**
   * Conference, Webinar,Trade Show,Public Relations, Partners, Referral Program,Advertisement,Banner Ads, Direct Mail,Email,Telemarketing, Others
   * @member {String} Type
   */
  exports.prototype['Type'] = undefined;

  /**
   * @member {String} Start Date
   */
  exports.prototype['Start Date'] = undefined;

  /**
   * @member {String} Expected Response
   */
  exports.prototype['Expected Response'] = undefined;

  /**
   * @member {String} Budgeted Cost
   */
  exports.prototype['Budgeted Cost'] = undefined;

  /**
   * @member {String} Campaign Owner
   */
  exports.prototype['Campaign Owner'] = undefined;

  /**
   * @member {String} Created By
   */
  exports.prototype['Created By'] = undefined;

  /**
   * @member {String} CAMPAIGNID
   */
  exports.prototype['CAMPAIGNID'] = undefined;

  /**
   * @member {String} Campaign Name
   */
  exports.prototype['Campaign Name'] = undefined;

  /**
   * @member {String} Num sent
   */
  exports.prototype['Num sent'] = undefined;




  return exports;
}));
