// zohocrm/src/model/Contact.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Zohocrm) {
      root.Zohocrm = {};
    }
    root.Zohocrm.Contact = factory(root.Zohocrm.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The Contact model module.
   * @module model/Contact
   * @version api-v2
   */

  /**
   * Constructs a new <code>Contact</code>.
   * @alias module:model/Contact
   * @class
   * @param lastName
   */
  var exports = function(lastName) {





















    this['Last Name'] = lastName;



























  };

  /**
   * Constructs a <code>Contact</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Contact} obj Optional instance to populate.
   * @return {module:model/Contact} The populated <code>Contact</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('ACCOUNTID')) {
        obj['ACCOUNTID'] = ApiClient.convertToType(data['ACCOUNTID'], 'String');
      }
      if (data.hasOwnProperty('Account Name')) {
        obj['Account Name'] = ApiClient.convertToType(data['Account Name'], 'String');
      }
      if (data.hasOwnProperty('Add to QuickBooks')) {
        obj['Add to QuickBooks'] = ApiClient.convertToType(data['Add to QuickBooks'], 'String');
      }
      if (data.hasOwnProperty('Assistant')) {
        obj['Assistant'] = ApiClient.convertToType(data['Assistant'], 'String');
      }
      if (data.hasOwnProperty('Asst Phone')) {
        obj['Asst Phone'] = ApiClient.convertToType(data['Asst Phone'], 'String');
      }
      if (data.hasOwnProperty('CAMPAIGNID')) {
        obj['CAMPAIGNID'] = ApiClient.convertToType(data['CAMPAIGNID'], 'String');
      }
      if (data.hasOwnProperty('CONTACTID')) {
        obj['CONTACTID'] = ApiClient.convertToType(data['CONTACTID'], 'String');
      }
      if (data.hasOwnProperty('Campaign Source')) {
        obj['Campaign Source'] = ApiClient.convertToType(data['Campaign Source'], 'String');
      }
      if (data.hasOwnProperty('Contact Owner')) {
        obj['Contact Owner'] = ApiClient.convertToType(data['Contact Owner'], 'String');
      }
      if (data.hasOwnProperty('Created By')) {
        obj['Created By'] = ApiClient.convertToType(data['Created By'], 'String');
      }
      if (data.hasOwnProperty('Created Time')) {
        obj['Created Time'] = ApiClient.convertToType(data['Created Time'], 'String');
      }
      if (data.hasOwnProperty('Date of Birth')) {
        obj['Date of Birth'] = ApiClient.convertToType(data['Date of Birth'], 'String');
      }
      if (data.hasOwnProperty('Department')) {
        obj['Department'] = ApiClient.convertToType(data['Department'], 'String');
      }
      if (data.hasOwnProperty('Description')) {
        obj['Description'] = ApiClient.convertToType(data['Description'], 'String');
      }
      if (data.hasOwnProperty('Email')) {
        obj['Email'] = ApiClient.convertToType(data['Email'], 'String');
      }
      if (data.hasOwnProperty('Email Opt Out')) {
        obj['Email Opt Out'] = ApiClient.convertToType(data['Email Opt Out'], 'String');
      }
      if (data.hasOwnProperty('Fax')) {
        obj['Fax'] = ApiClient.convertToType(data['Fax'], 'String');
      }
      if (data.hasOwnProperty('First Name')) {
        obj['First Name'] = ApiClient.convertToType(data['First Name'], 'String');
      }
      if (data.hasOwnProperty('Home Phone')) {
        obj['Home Phone'] = ApiClient.convertToType(data['Home Phone'], 'String');
      }
      if (data.hasOwnProperty('Last Activity Time')) {
        obj['Last Activity Time'] = ApiClient.convertToType(data['Last Activity Time'], 'String');
      }
      if (data.hasOwnProperty('Last Name')) {
        obj['Last Name'] = ApiClient.convertToType(data['Last Name'], 'String');
      }
      if (data.hasOwnProperty('Lead Source')) {
        obj['Lead Source'] = ApiClient.convertToType(data['Lead Source'], 'String');
      }
      if (data.hasOwnProperty('MODIFIEDBY')) {
        obj['MODIFIEDBY'] = ApiClient.convertToType(data['MODIFIEDBY'], 'String');
      }
      if (data.hasOwnProperty('Mailing City')) {
        obj['Mailing City'] = ApiClient.convertToType(data['Mailing City'], 'String');
      }
      if (data.hasOwnProperty('Mailing Country')) {
        obj['Mailing Country'] = ApiClient.convertToType(data['Mailing Country'], 'String');
      }
      if (data.hasOwnProperty('Mailing State')) {
        obj['Mailing State'] = ApiClient.convertToType(data['Mailing State'], 'String');
      }
      if (data.hasOwnProperty('Mailing Street')) {
        obj['Mailing Street'] = ApiClient.convertToType(data['Mailing Street'], 'String');
      }
      if (data.hasOwnProperty('Mailing Zip')) {
        obj['Mailing Zip'] = ApiClient.convertToType(data['Mailing Zip'], 'String');
      }
      if (data.hasOwnProperty('Mobile')) {
        obj['Mobile'] = ApiClient.convertToType(data['Mobile'], 'String');
      }
      if (data.hasOwnProperty('Modified By')) {
        obj['Modified By'] = ApiClient.convertToType(data['Modified By'], 'String');
      }
      if (data.hasOwnProperty('Modified Time')) {
        obj['Modified Time'] = ApiClient.convertToType(data['Modified Time'], 'String');
      }
      if (data.hasOwnProperty('Other City')) {
        obj['Other City'] = ApiClient.convertToType(data['Other City'], 'String');
      }
      if (data.hasOwnProperty('Other Country')) {
        obj['Other Country'] = ApiClient.convertToType(data['Other Country'], 'String');
      }
      if (data.hasOwnProperty('Other Phone')) {
        obj['Other Phone'] = ApiClient.convertToType(data['Other Phone'], 'String');
      }
      if (data.hasOwnProperty('Other State')) {
        obj['Other State'] = ApiClient.convertToType(data['Other State'], 'String');
      }
      if (data.hasOwnProperty('Other Street')) {
        obj['Other Street'] = ApiClient.convertToType(data['Other Street'], 'String');
      }
      if (data.hasOwnProperty('Other Zip')) {
        obj['Other Zip'] = ApiClient.convertToType(data['Other Zip'], 'String');
      }
      if (data.hasOwnProperty('Phone')) {
        obj['Phone'] = ApiClient.convertToType(data['Phone'], 'String');
      }
      if (data.hasOwnProperty('Reports To')) {
        obj['Reports To'] = ApiClient.convertToType(data['Reports To'], 'String');
      }
      if (data.hasOwnProperty('SMCREATORID')) {
        obj['SMCREATORID'] = ApiClient.convertToType(data['SMCREATORID'], 'String');
      }
      if (data.hasOwnProperty('SMOWNERID')) {
        obj['SMOWNERID'] = ApiClient.convertToType(data['SMOWNERID'], 'String');
      }
      if (data.hasOwnProperty('Salutation')) {
        obj['Salutation'] = ApiClient.convertToType(data['Salutation'], 'String');
      }
      if (data.hasOwnProperty('Secondary Email')) {
        obj['Secondary Email'] = ApiClient.convertToType(data['Secondary Email'], 'String');
      }
      if (data.hasOwnProperty('Skype ID')) {
        obj['Skype ID'] = ApiClient.convertToType(data['Skype ID'], 'String');
      }
      if (data.hasOwnProperty('Title')) {
        obj['Title'] = ApiClient.convertToType(data['Title'], 'String');
      }
      if (data.hasOwnProperty('Twitter')) {
        obj['Twitter'] = ApiClient.convertToType(data['Twitter'], 'String');
      }
      if (data.hasOwnProperty('VENDORID')) {
        obj['VENDORID'] = ApiClient.convertToType(data['VENDORID'], 'String');
      }
      if (data.hasOwnProperty('Vendor Name')) {
        obj['Vendor Name'] = ApiClient.convertToType(data['Vendor Name'], 'String');
      }
    }
    return obj;
  }


  /**
   * @member {String} ACCOUNTID
   */
  exports.prototype['ACCOUNTID'] = undefined;

  /**
   * @member {String} Account Name
   */
  exports.prototype['Account Name'] = undefined;

  /**
   * @member {String} Add to QuickBooks
   */
  exports.prototype['Add to QuickBooks'] = undefined;

  /**
   * @member {String} Assistant
   */
  exports.prototype['Assistant'] = undefined;

  /**
   * @member {String} Asst Phone
   */
  exports.prototype['Asst Phone'] = undefined;

  /**
   * @member {String} CAMPAIGNID
   */
  exports.prototype['CAMPAIGNID'] = undefined;

  /**
   * @member {String} CONTACTID
   */
  exports.prototype['CONTACTID'] = undefined;

  /**
   * @member {String} Campaign Source
   */
  exports.prototype['Campaign Source'] = undefined;

  /**
   * @member {String} Contact Owner
   */
  exports.prototype['Contact Owner'] = undefined;

  /**
   * @member {String} Created By
   */
  exports.prototype['Created By'] = undefined;

  /**
   * @member {String} Created Time
   */
  exports.prototype['Created Time'] = undefined;

  /**
   * @member {String} Date of Birth
   */
  exports.prototype['Date of Birth'] = undefined;

  /**
   * @member {String} Department
   */
  exports.prototype['Department'] = undefined;

  /**
   * @member {String} Description
   */
  exports.prototype['Description'] = undefined;

  /**
   * @member {String} Email
   */
  exports.prototype['Email'] = undefined;

  /**
   * @member {String} Email Opt Out
   */
  exports.prototype['Email Opt Out'] = undefined;

  /**
   * @member {String} Fax
   */
  exports.prototype['Fax'] = undefined;

  /**
   * @member {String} First Name
   */
  exports.prototype['First Name'] = undefined;

  /**
   * @member {String} Home Phone
   */
  exports.prototype['Home Phone'] = undefined;

  /**
   * @member {String} Last Activity Time
   */
  exports.prototype['Last Activity Time'] = undefined;

  /**
   * @member {String} Last Name
   */
  exports.prototype['Last Name'] = undefined;

  /**
   * @member {String} Lead Source
   */
  exports.prototype['Lead Source'] = undefined;

  /**
   * @member {String} MODIFIEDBY
   */
  exports.prototype['MODIFIEDBY'] = undefined;

  /**
   * @member {String} Mailing City
   */
  exports.prototype['Mailing City'] = undefined;

  /**
   * @member {String} Mailing Country
   */
  exports.prototype['Mailing Country'] = undefined;

  /**
   * @member {String} Mailing State
   */
  exports.prototype['Mailing State'] = undefined;

  /**
   * @member {String} Mailing Street
   */
  exports.prototype['Mailing Street'] = undefined;

  /**
   * @member {String} Mailing Zip
   */
  exports.prototype['Mailing Zip'] = undefined;

  /**
   * @member {String} Mobile
   */
  exports.prototype['Mobile'] = undefined;

  /**
   * @member {String} Modified By
   */
  exports.prototype['Modified By'] = undefined;

  /**
   * @member {String} Modified Time
   */
  exports.prototype['Modified Time'] = undefined;

  /**
   * @member {String} Other City
   */
  exports.prototype['Other City'] = undefined;

  /**
   * @member {String} Other Country
   */
  exports.prototype['Other Country'] = undefined;

  /**
   * @member {String} Other Phone
   */
  exports.prototype['Other Phone'] = undefined;

  /**
   * @member {String} Other State
   */
  exports.prototype['Other State'] = undefined;

  /**
   * @member {String} Other Street
   */
  exports.prototype['Other Street'] = undefined;

  /**
   * @member {String} Other Zip
   */
  exports.prototype['Other Zip'] = undefined;

  /**
   * @member {String} Phone
   */
  exports.prototype['Phone'] = undefined;

  /**
   * @member {String} Reports To
   */
  exports.prototype['Reports To'] = undefined;

  /**
   * @member {String} SMCREATORID
   */
  exports.prototype['SMCREATORID'] = undefined;

  /**
   * @member {String} SMOWNERID
   */
  exports.prototype['SMOWNERID'] = undefined;

  /**
   * @member {String} Salutation
   */
  exports.prototype['Salutation'] = undefined;

  /**
   * @member {String} Secondary Email
   */
  exports.prototype['Secondary Email'] = undefined;

  /**
   * @member {String} Skype ID
   */
  exports.prototype['Skype ID'] = undefined;

  /**
   * @member {String} Title
   */
  exports.prototype['Title'] = undefined;

  /**
   * @member {String} Twitter
   */
  exports.prototype['Twitter'] = undefined;

  /**
   * @member {String} VENDORID
   */
  exports.prototype['VENDORID'] = undefined;

  /**
   * @member {String} Vendor Name
   */
  exports.prototype['Vendor Name'] = undefined;




  return exports;
}));
