// zohocrm/src/model/Events.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Zohocrm) {
      root.Zohocrm = {};
    }
    root.Zohocrm.Events = factory(root.Zohocrm.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The Events model module.
   * @module model/Events
   * @version api-v2
   */

  /**
   * Constructs a new <code>Events</code>.
   * @alias module:model/Events
   * @class
   */
  var exports = function() {




















  };

  /**
   * Constructs a <code>Events</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Events} obj Optional instance to populate.
   * @return {module:model/Events} The populated <code>Events</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('ACTIVITYID')) {
        obj['ACTIVITYID'] = ApiClient.convertToType(data['ACTIVITYID'], 'String');
      }
      if (data.hasOwnProperty('CONTACTID')) {
        obj['CONTACTID'] = ApiClient.convertToType(data['CONTACTID'], 'String');
      }
      if (data.hasOwnProperty('Contact Name')) {
        obj['Contact Name'] = ApiClient.convertToType(data['Contact Name'], 'String');
      }
      if (data.hasOwnProperty('Created By')) {
        obj['Created By'] = ApiClient.convertToType(data['Created By'], 'String');
      }
      if (data.hasOwnProperty('Created Time')) {
        obj['Created Time'] = ApiClient.convertToType(data['Created Time'], 'Date');
      }
      if (data.hasOwnProperty('End DateTime')) {
        obj['End DateTime'] = ApiClient.convertToType(data['End DateTime'], 'Date');
      }
      if (data.hasOwnProperty('Event Owner')) {
        obj['Event Owner'] = ApiClient.convertToType(data['Event Owner'], 'String');
      }
      if (data.hasOwnProperty('MODIFIEDBY')) {
        obj['MODIFIEDBY'] = ApiClient.convertToType(data['MODIFIEDBY'], 'String');
      }
      if (data.hasOwnProperty('Modified By')) {
        obj['Modified By'] = ApiClient.convertToType(data['Modified By'], 'String');
      }
      if (data.hasOwnProperty('Modified Time')) {
        obj['Modified Time'] = ApiClient.convertToType(data['Modified Time'], 'Date');
      }
      if (data.hasOwnProperty('RELATEDTOID')) {
        obj['RELATEDTOID'] = ApiClient.convertToType(data['RELATEDTOID'], 'String');
      }
      if (data.hasOwnProperty('Related To')) {
        obj['Related To'] = ApiClient.convertToType(data['Related To'], 'String');
      }
      if (data.hasOwnProperty('SEMODULE')) {
        obj['SEMODULE'] = ApiClient.convertToType(data['SEMODULE'], 'String');
      }
      if (data.hasOwnProperty('SMCREATORID')) {
        obj['SMCREATORID'] = ApiClient.convertToType(data['SMCREATORID'], 'String');
      }
      if (data.hasOwnProperty('SMOWNERID')) {
        obj['SMOWNERID'] = ApiClient.convertToType(data['SMOWNERID'], 'String');
      }
      if (data.hasOwnProperty('Start DateTime')) {
        obj['Start DateTime'] = ApiClient.convertToType(data['Start DateTime'], 'Date');
      }
      if (data.hasOwnProperty('Subject')) {
        obj['Subject'] = ApiClient.convertToType(data['Subject'], 'String');
      }
      if (data.hasOwnProperty('UID')) {
        obj['UID'] = ApiClient.convertToType(data['UID'], 'String');
      }
      if (data.hasOwnProperty('Venue')) {
        obj['Venue'] = ApiClient.convertToType(data['Venue'], 'String');
      }
    }
    return obj;
  }


  /**
   * @member {String} ACTIVITYID
   */
  exports.prototype['ACTIVITYID'] = undefined;

  /**
   * @member {String} CONTACTID
   */
  exports.prototype['CONTACTID'] = undefined;

  /**
   * @member {String} Contact Name
   */
  exports.prototype['Contact Name'] = undefined;

  /**
   * @member {String} Created By
   */
  exports.prototype['Created By'] = undefined;

  /**
   * @member {Date} Created Time
   */
  exports.prototype['Created Time'] = undefined;

  /**
   * @member {Date} End DateTime
   */
  exports.prototype['End DateTime'] = undefined;

  /**
   * @member {String} Event Owner
   */
  exports.prototype['Event Owner'] = undefined;

  /**
   * @member {String} MODIFIEDBY
   */
  exports.prototype['MODIFIEDBY'] = undefined;

  /**
   * @member {String} Modified By
   */
  exports.prototype['Modified By'] = undefined;

  /**
   * @member {Date} Modified Time
   */
  exports.prototype['Modified Time'] = undefined;

  /**
   * @member {String} RELATEDTOID
   */
  exports.prototype['RELATEDTOID'] = undefined;

  /**
   * @member {String} Related To
   */
  exports.prototype['Related To'] = undefined;

  /**
   * @member {String} SEMODULE
   */
  exports.prototype['SEMODULE'] = undefined;

  /**
   * @member {String} SMCREATORID
   */
  exports.prototype['SMCREATORID'] = undefined;

  /**
   * @member {String} SMOWNERID
   */
  exports.prototype['SMOWNERID'] = undefined;

  /**
   * @member {Date} Start DateTime
   */
  exports.prototype['Start DateTime'] = undefined;

  /**
   * @member {String} Subject
   */
  exports.prototype['Subject'] = undefined;

  /**
   * @member {String} UID
   */
  exports.prototype['UID'] = undefined;

  /**
   * @member {String} Venue
   */
  exports.prototype['Venue'] = undefined;




  return exports;
}));
