// zohocrm/src/model/Lead.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Zohocrm) {
      root.Zohocrm = {};
    }
    root.Zohocrm.Lead = factory(root.Zohocrm.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The Lead model module.
   * @module model/Lead
   * @version api-v2
   */

  /**
   * Constructs a new <code>Lead</code>.
   * @alias module:model/Lead
   * @class
   * @param lastName
   */
  var exports = function(lastName) {


















    this['Last Name'] = lastName;




















  };

  /**
   * Constructs a <code>Lead</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Lead} obj Optional instance to populate.
   * @return {module:model/Lead} The populated <code>Lead</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('Annual Revenue')) {
        obj['Annual Revenue'] = ApiClient.convertToType(data['Annual Revenue'], 'String');
      }
      if (data.hasOwnProperty('CAMPAIGNID')) {
        obj['CAMPAIGNID'] = ApiClient.convertToType(data['CAMPAIGNID'], 'String');
      }
      if (data.hasOwnProperty('Campaign Source')) {
        obj['Campaign Source'] = ApiClient.convertToType(data['Campaign Source'], 'String');
      }
      if (data.hasOwnProperty('City')) {
        obj['City'] = ApiClient.convertToType(data['City'], 'String');
      }
      if (data.hasOwnProperty('Company')) {
        obj['Company'] = ApiClient.convertToType(data['Company'], 'String');
      }
      if (data.hasOwnProperty('Country')) {
        obj['Country'] = ApiClient.convertToType(data['Country'], 'String');
      }
      if (data.hasOwnProperty('Created By')) {
        obj['Created By'] = ApiClient.convertToType(data['Created By'], 'String');
      }
      if (data.hasOwnProperty('Created Time')) {
        obj['Created Time'] = ApiClient.convertToType(data['Created Time'], 'String');
      }
      if (data.hasOwnProperty('Description')) {
        obj['Description'] = ApiClient.convertToType(data['Description'], 'String');
      }
      if (data.hasOwnProperty('Designation')) {
        obj['Designation'] = ApiClient.convertToType(data['Designation'], 'String');
      }
      if (data.hasOwnProperty('Email')) {
        obj['Email'] = ApiClient.convertToType(data['Email'], 'String');
      }
      if (data.hasOwnProperty('Email Opt Out')) {
        obj['Email Opt Out'] = ApiClient.convertToType(data['Email Opt Out'], 'String');
      }
      if (data.hasOwnProperty('Fax')) {
        obj['Fax'] = ApiClient.convertToType(data['Fax'], 'String');
      }
      if (data.hasOwnProperty('First Name')) {
        obj['First Name'] = ApiClient.convertToType(data['First Name'], 'String');
      }
      if (data.hasOwnProperty('Industry')) {
        obj['Industry'] = ApiClient.convertToType(data['Industry'], 'String');
      }
      if (data.hasOwnProperty('LEADID')) {
        obj['LEADID'] = ApiClient.convertToType(data['LEADID'], 'String');
      }
      if (data.hasOwnProperty('Last Activity Time')) {
        obj['Last Activity Time'] = ApiClient.convertToType(data['Last Activity Time'], 'String');
      }
      if (data.hasOwnProperty('Last Name')) {
        obj['Last Name'] = ApiClient.convertToType(data['Last Name'], 'String');
      }
      if (data.hasOwnProperty('Lead Owner')) {
        obj['Lead Owner'] = ApiClient.convertToType(data['Lead Owner'], 'String');
      }
      if (data.hasOwnProperty('Lead Source')) {
        obj['Lead Source'] = ApiClient.convertToType(data['Lead Source'], 'String');
      }
      if (data.hasOwnProperty('Lead Status')) {
        obj['Lead Status'] = ApiClient.convertToType(data['Lead Status'], 'String');
      }
      if (data.hasOwnProperty('MODIFIEDBY')) {
        obj['MODIFIEDBY'] = ApiClient.convertToType(data['MODIFIEDBY'], 'String');
      }
      if (data.hasOwnProperty('Mobile')) {
        obj['Mobile'] = ApiClient.convertToType(data['Mobile'], 'String');
      }
      if (data.hasOwnProperty('Modified By')) {
        obj['Modified By'] = ApiClient.convertToType(data['Modified By'], 'String');
      }
      if (data.hasOwnProperty('Modified Time')) {
        obj['Modified Time'] = ApiClient.convertToType(data['Modified Time'], 'String');
      }
      if (data.hasOwnProperty('No of Employees')) {
        obj['No of Employees'] = ApiClient.convertToType(data['No of Employees'], 'String');
      }
      if (data.hasOwnProperty('Phone')) {
        obj['Phone'] = ApiClient.convertToType(data['Phone'], 'String');
      }
      if (data.hasOwnProperty('Rating')) {
        obj['Rating'] = ApiClient.convertToType(data['Rating'], 'String');
      }
      if (data.hasOwnProperty('SMCREATORID')) {
        obj['SMCREATORID'] = ApiClient.convertToType(data['SMCREATORID'], 'String');
      }
      if (data.hasOwnProperty('SMOWNERID')) {
        obj['SMOWNERID'] = ApiClient.convertToType(data['SMOWNERID'], 'String');
      }
      if (data.hasOwnProperty('Salutation')) {
        obj['Salutation'] = ApiClient.convertToType(data['Salutation'], 'String');
      }
      if (data.hasOwnProperty('Secondary Email')) {
        obj['Secondary Email'] = ApiClient.convertToType(data['Secondary Email'], 'String');
      }
      if (data.hasOwnProperty('Skype ID')) {
        obj['Skype ID'] = ApiClient.convertToType(data['Skype ID'], 'String');
      }
      if (data.hasOwnProperty('State')) {
        obj['State'] = ApiClient.convertToType(data['State'], 'String');
      }
      if (data.hasOwnProperty('Street')) {
        obj['Street'] = ApiClient.convertToType(data['Street'], 'String');
      }
      if (data.hasOwnProperty('Twitter')) {
        obj['Twitter'] = ApiClient.convertToType(data['Twitter'], 'String');
      }
      if (data.hasOwnProperty('Website')) {
        obj['Website'] = ApiClient.convertToType(data['Website'], 'String');
      }
      if (data.hasOwnProperty('Zip Code')) {
        obj['Zip Code'] = ApiClient.convertToType(data['Zip Code'], 'String');
      }
    }
    return obj;
  }


  /**
   * @member {String} Annual Revenue
   */
  exports.prototype['Annual Revenue'] = undefined;

  /**
   * @member {String} CAMPAIGNID
   */
  exports.prototype['CAMPAIGNID'] = undefined;

  /**
   * @member {String} Campaign Source
   */
  exports.prototype['Campaign Source'] = undefined;

  /**
   * @member {String} City
   */
  exports.prototype['City'] = undefined;

  /**
   * @member {String} Company
   */
  exports.prototype['Company'] = undefined;

  /**
   * @member {String} Country
   */
  exports.prototype['Country'] = undefined;

  /**
   * @member {String} Created By
   */
  exports.prototype['Created By'] = undefined;

  /**
   * @member {String} Created Time
   */
  exports.prototype['Created Time'] = undefined;

  /**
   * @member {String} Description
   */
  exports.prototype['Description'] = undefined;

  /**
   * @member {String} Designation
   */
  exports.prototype['Designation'] = undefined;

  /**
   * @member {String} Email
   */
  exports.prototype['Email'] = undefined;

  /**
   * @member {String} Email Opt Out
   */
  exports.prototype['Email Opt Out'] = undefined;

  /**
   * @member {String} Fax
   */
  exports.prototype['Fax'] = undefined;

  /**
   * @member {String} First Name
   */
  exports.prototype['First Name'] = undefined;

  /**
   * @member {String} Industry
   */
  exports.prototype['Industry'] = undefined;

  /**
   * @member {String} LEADID
   */
  exports.prototype['LEADID'] = undefined;

  /**
   * @member {String} Last Activity Time
   */
  exports.prototype['Last Activity Time'] = undefined;

  /**
   * @member {String} Last Name
   */
  exports.prototype['Last Name'] = undefined;

  /**
   * @member {String} Lead Owner
   */
  exports.prototype['Lead Owner'] = undefined;

  /**
   * @member {String} Lead Source
   */
  exports.prototype['Lead Source'] = undefined;

  /**
   * @member {String} Lead Status
   */
  exports.prototype['Lead Status'] = undefined;

  /**
   * @member {String} MODIFIEDBY
   */
  exports.prototype['MODIFIEDBY'] = undefined;

  /**
   * @member {String} Mobile
   */
  exports.prototype['Mobile'] = undefined;

  /**
   * @member {String} Modified By
   */
  exports.prototype['Modified By'] = undefined;

  /**
   * @member {String} Modified Time
   */
  exports.prototype['Modified Time'] = undefined;

  /**
   * @member {String} No of Employees
   */
  exports.prototype['No of Employees'] = undefined;

  /**
   * @member {String} Phone
   */
  exports.prototype['Phone'] = undefined;

  /**
   * @member {String} Rating
   */
  exports.prototype['Rating'] = undefined;

  /**
   * @member {String} SMCREATORID
   */
  exports.prototype['SMCREATORID'] = undefined;

  /**
   * @member {String} SMOWNERID
   */
  exports.prototype['SMOWNERID'] = undefined;

  /**
   * @member {String} Salutation
   */
  exports.prototype['Salutation'] = undefined;

  /**
   * @member {String} Secondary Email
   */
  exports.prototype['Secondary Email'] = undefined;

  /**
   * @member {String} Skype ID
   */
  exports.prototype['Skype ID'] = undefined;

  /**
   * @member {String} State
   */
  exports.prototype['State'] = undefined;

  /**
   * @member {String} Street
   */
  exports.prototype['Street'] = undefined;

  /**
   * @member {String} Twitter
   */
  exports.prototype['Twitter'] = undefined;

  /**
   * @member {String} Website
   */
  exports.prototype['Website'] = undefined;

  /**
   * @member {String} Zip Code
   */
  exports.prototype['Zip Code'] = undefined;




  return exports;
}));
