// zohocrm/src/model/Notes.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Zohocrm) {
      root.Zohocrm = {};
    }
    root.Zohocrm.Notes = factory(root.Zohocrm.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The Notes model module.
   * @module model/Notes
   * @version api-v2
   */

  /**
   * Constructs a new <code>Notes</code>.
   * @alias module:model/Notes
   * @class
   */
  var exports = function() {













  };

  /**
   * Constructs a <code>Notes</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Notes} obj Optional instance to populate.
   * @return {module:model/Notes} The populated <code>Notes</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('Created By')) {
        obj['Created By'] = ApiClient.convertToType(data['Created By'], 'String');
      }
      if (data.hasOwnProperty('Created Time')) {
        obj['Created Time'] = ApiClient.convertToType(data['Created Time'], 'Date');
      }
      if (data.hasOwnProperty('ISVOICENOTES')) {
        obj['ISVOICENOTES'] = ApiClient.convertToType(data['ISVOICENOTES'], 'Boolean');
      }
      if (data.hasOwnProperty('MODIFIEDBY')) {
        obj['MODIFIEDBY'] = ApiClient.convertToType(data['MODIFIEDBY'], 'String');
      }
      if (data.hasOwnProperty('Modified By')) {
        obj['Modified By'] = ApiClient.convertToType(data['Modified By'], 'String');
      }
      if (data.hasOwnProperty('Modified Time')) {
        obj['Modified Time'] = ApiClient.convertToType(data['Modified Time'], 'Date');
      }
      if (data.hasOwnProperty('Note Content')) {
        obj['Note Content'] = ApiClient.convertToType(data['Note Content'], 'String');
      }
      if (data.hasOwnProperty('Owner Name')) {
        obj['Owner Name'] = ApiClient.convertToType(data['Owner Name'], 'String');
      }
      if (data.hasOwnProperty('SMCREATORID')) {
        obj['SMCREATORID'] = ApiClient.convertToType(data['SMCREATORID'], 'String');
      }
      if (data.hasOwnProperty('SMOWNERID')) {
        obj['SMOWNERID'] = ApiClient.convertToType(data['SMOWNERID'], 'String');
      }
      if (data.hasOwnProperty('Title')) {
        obj['Title'] = ApiClient.convertToType(data['Title'], 'String');
      }
      if (data.hasOwnProperty('id')) {
        obj['id'] = ApiClient.convertToType(data['id'], 'String');
      }
    }
    return obj;
  }


  /**
   * @member {String} Created By
   */
  exports.prototype['Created By'] = undefined;

  /**
   * @member {Date} Created Time
   */
  exports.prototype['Created Time'] = undefined;

  /**
   * @member {Boolean} ISVOICENOTES
   */
  exports.prototype['ISVOICENOTES'] = undefined;

  /**
   * @member {String} MODIFIEDBY
   */
  exports.prototype['MODIFIEDBY'] = undefined;

  /**
   * @member {String} Modified By
   */
  exports.prototype['Modified By'] = undefined;

  /**
   * @member {Date} Modified Time
   */
  exports.prototype['Modified Time'] = undefined;

  /**
   * @member {String} Note Content
   */
  exports.prototype['Note Content'] = undefined;

  /**
   * @member {String} Owner Name
   */
  exports.prototype['Owner Name'] = undefined;

  /**
   * @member {String} SMCREATORID
   */
  exports.prototype['SMCREATORID'] = undefined;

  /**
   * @member {String} SMOWNERID
   */
  exports.prototype['SMOWNERID'] = undefined;

  /**
   * @member {String} Title
   */
  exports.prototype['Title'] = undefined;

  /**
   * @member {String} id
   */
  exports.prototype['id'] = undefined;




  return exports;
}));
