// zohocrm/src/model/Opportunity.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Zohocrm) {
      root.Zohocrm = {};
    }
    root.Zohocrm.Opportunity = factory(root.Zohocrm.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The Opportunity model module.
   * @module model/Opportunity
   * @version api-v2
   */

  /**
   * Constructs a new <code>Opportunity</code>.
   * @alias module:model/Opportunity
   * @class
   * @param potentialName
   */
  var exports = function(potentialName) {






















    this['Potential Name'] = potentialName;







  };

  /**
   * Constructs a <code>Opportunity</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Opportunity} obj Optional instance to populate.
   * @return {module:model/Opportunity} The populated <code>Opportunity</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('ACCOUNTID')) {
        obj['ACCOUNTID'] = ApiClient.convertToType(data['ACCOUNTID'], 'String');
      }
      if (data.hasOwnProperty('Account Name')) {
        obj['Account Name'] = ApiClient.convertToType(data['Account Name'], 'String');
      }
      if (data.hasOwnProperty('Amount')) {
        obj['Amount'] = ApiClient.convertToType(data['Amount'], 'String');
      }
      if (data.hasOwnProperty('CAMPAIGNID')) {
        obj['CAMPAIGNID'] = ApiClient.convertToType(data['CAMPAIGNID'], 'String');
      }
      if (data.hasOwnProperty('CONTACTID')) {
        obj['CONTACTID'] = ApiClient.convertToType(data['CONTACTID'], 'String');
      }
      if (data.hasOwnProperty('Campaign Source')) {
        obj['Campaign Source'] = ApiClient.convertToType(data['Campaign Source'], 'String');
      }
      if (data.hasOwnProperty('Closing Date')) {
        obj['Closing Date'] = ApiClient.convertToType(data['Closing Date'], 'String');
      }
      if (data.hasOwnProperty('Contact Name')) {
        obj['Contact Name'] = ApiClient.convertToType(data['Contact Name'], 'String');
      }
      if (data.hasOwnProperty('Created By')) {
        obj['Created By'] = ApiClient.convertToType(data['Created By'], 'String');
      }
      if (data.hasOwnProperty('Created Time')) {
        obj['Created Time'] = ApiClient.convertToType(data['Created Time'], 'String');
      }
      if (data.hasOwnProperty('Description')) {
        obj['Description'] = ApiClient.convertToType(data['Description'], 'String');
      }
      if (data.hasOwnProperty('Expected Revenue')) {
        obj['Expected Revenue'] = ApiClient.convertToType(data['Expected Revenue'], 'String');
      }
      if (data.hasOwnProperty('Last Activity Time')) {
        obj['Last Activity Time'] = ApiClient.convertToType(data['Last Activity Time'], 'String');
      }
      if (data.hasOwnProperty('Lead Conversion Time')) {
        obj['Lead Conversion Time'] = ApiClient.convertToType(data['Lead Conversion Time'], 'String');
      }
      if (data.hasOwnProperty('Lead Source')) {
        obj['Lead Source'] = ApiClient.convertToType(data['Lead Source'], 'String');
      }
      if (data.hasOwnProperty('MODIFIEDBY')) {
        obj['MODIFIEDBY'] = ApiClient.convertToType(data['MODIFIEDBY'], 'String');
      }
      if (data.hasOwnProperty('Modified By')) {
        obj['Modified By'] = ApiClient.convertToType(data['Modified By'], 'String');
      }
      if (data.hasOwnProperty('Modified Time')) {
        obj['Modified Time'] = ApiClient.convertToType(data['Modified Time'], 'String');
      }
      if (data.hasOwnProperty('Next Step')) {
        obj['Next Step'] = ApiClient.convertToType(data['Next Step'], 'String');
      }
      if (data.hasOwnProperty('Overall Sales Duration')) {
        obj['Overall Sales Duration'] = ApiClient.convertToType(data['Overall Sales Duration'], 'String');
      }
      if (data.hasOwnProperty('POTENTIALID')) {
        obj['POTENTIALID'] = ApiClient.convertToType(data['POTENTIALID'], 'String');
      }
      if (data.hasOwnProperty('Potential Name')) {
        obj['Potential Name'] = ApiClient.convertToType(data['Potential Name'], 'String');
      }
      if (data.hasOwnProperty('Potential Owner')) {
        obj['Potential Owner'] = ApiClient.convertToType(data['Potential Owner'], 'String');
      }
      if (data.hasOwnProperty('Probability')) {
        obj['Probability'] = ApiClient.convertToType(data['Probability'], 'String');
      }
      if (data.hasOwnProperty('SMCREATORID')) {
        obj['SMCREATORID'] = ApiClient.convertToType(data['SMCREATORID'], 'String');
      }
      if (data.hasOwnProperty('SMOWNERID')) {
        obj['SMOWNERID'] = ApiClient.convertToType(data['SMOWNERID'], 'String');
      }
      if (data.hasOwnProperty('Sales Cycle Duration')) {
        obj['Sales Cycle Duration'] = ApiClient.convertToType(data['Sales Cycle Duration'], 'String');
      }
      if (data.hasOwnProperty('Stage')) {
        obj['Stage'] = ApiClient.convertToType(data['Stage'], 'String');
      }
      if (data.hasOwnProperty('Type')) {
        obj['Type'] = ApiClient.convertToType(data['Type'], 'String');
      }
    }
    return obj;
  }


  /**
   * @member {String} ACCOUNTID
   */
  exports.prototype['ACCOUNTID'] = undefined;

  /**
   * @member {String} Account Name
   */
  exports.prototype['Account Name'] = undefined;

  /**
   * @member {String} Amount
   */
  exports.prototype['Amount'] = undefined;

  /**
   * @member {String} CAMPAIGNID
   */
  exports.prototype['CAMPAIGNID'] = undefined;

  /**
   * @member {String} CONTACTID
   */
  exports.prototype['CONTACTID'] = undefined;

  /**
   * @member {String} Campaign Source
   */
  exports.prototype['Campaign Source'] = undefined;

  /**
   * @member {String} Closing Date
   */
  exports.prototype['Closing Date'] = undefined;

  /**
   * @member {String} Contact Name
   */
  exports.prototype['Contact Name'] = undefined;

  /**
   * @member {String} Created By
   */
  exports.prototype['Created By'] = undefined;

  /**
   * @member {String} Created Time
   */
  exports.prototype['Created Time'] = undefined;

  /**
   * @member {String} Description
   */
  exports.prototype['Description'] = undefined;

  /**
   * @member {String} Expected Revenue
   */
  exports.prototype['Expected Revenue'] = undefined;

  /**
   * @member {String} Last Activity Time
   */
  exports.prototype['Last Activity Time'] = undefined;

  /**
   * @member {String} Lead Conversion Time
   */
  exports.prototype['Lead Conversion Time'] = undefined;

  /**
   * @member {String} Lead Source
   */
  exports.prototype['Lead Source'] = undefined;

  /**
   * @member {String} MODIFIEDBY
   */
  exports.prototype['MODIFIEDBY'] = undefined;

  /**
   * @member {String} Modified By
   */
  exports.prototype['Modified By'] = undefined;

  /**
   * @member {String} Modified Time
   */
  exports.prototype['Modified Time'] = undefined;

  /**
   * @member {String} Next Step
   */
  exports.prototype['Next Step'] = undefined;

  /**
   * @member {String} Overall Sales Duration
   */
  exports.prototype['Overall Sales Duration'] = undefined;

  /**
   * @member {String} POTENTIALID
   */
  exports.prototype['POTENTIALID'] = undefined;

  /**
   * @member {String} Potential Name
   */
  exports.prototype['Potential Name'] = undefined;

  /**
   * @member {String} Potential Owner
   */
  exports.prototype['Potential Owner'] = undefined;

  /**
   * @member {String} Probability
   */
  exports.prototype['Probability'] = undefined;

  /**
   * @member {String} SMCREATORID
   */
  exports.prototype['SMCREATORID'] = undefined;

  /**
   * @member {String} SMOWNERID
   */
  exports.prototype['SMOWNERID'] = undefined;

  /**
   * @member {String} Sales Cycle Duration
   */
  exports.prototype['Sales Cycle Duration'] = undefined;

  /**
   * @member {String} Stage
   */
  exports.prototype['Stage'] = undefined;

  /**
   * @member {String} Type
   */
  exports.prototype['Type'] = undefined;




  return exports;
}));
