// zohocrm/src/model/Tasks.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.Zohocrm) {
      root.Zohocrm = {};
    }
    root.Zohocrm.Tasks = factory(root.Zohocrm.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The Tasks model module.
   * @module model/Tasks
   * @version api-v2
   */

  /**
   * Constructs a new <code>Tasks</code>.
   * @alias module:model/Tasks
   * @class
   */
  var exports = function() {












  };

  /**
   * Constructs a <code>Tasks</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/Tasks} obj Optional instance to populate.
   * @return {module:model/Tasks} The populated <code>Tasks</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('Accounts')) {
        obj['Accounts'] = ApiClient.convertToType(data['Accounts'], 'String');
      }
      if (data.hasOwnProperty('Contacts/Leads')) {
        obj['Contacts/Leads'] = ApiClient.convertToType(data['Contacts/Leads'], 'String');
      }
      if (data.hasOwnProperty('Description')) {
        obj['Description'] = ApiClient.convertToType(data['Description'], 'String');
      }
      if (data.hasOwnProperty('Due Date')) {
        obj['Due Date'] = ApiClient.convertToType(data['Due Date'], 'Date');
      }
      if (data.hasOwnProperty('Priority')) {
        obj['Priority'] = ApiClient.convertToType(data['Priority'], 'String');
      }
      if (data.hasOwnProperty('Recurring Activity')) {
        obj['Recurring Activity'] = ApiClient.convertToType(data['Recurring Activity'], 'Boolean');
      }
      if (data.hasOwnProperty('Remind At')) {
        obj['Remind At'] = ApiClient.convertToType(data['Remind At'], 'Boolean');
      }
      if (data.hasOwnProperty('Send Notification Email')) {
        obj['Send Notification Email'] = ApiClient.convertToType(data['Send Notification Email'], 'Boolean');
      }
      if (data.hasOwnProperty('Status')) {
        obj['Status'] = ApiClient.convertToType(data['Status'], 'String');
      }
      if (data.hasOwnProperty('Subject')) {
        obj['Subject'] = ApiClient.convertToType(data['Subject'], 'String');
      }
      if (data.hasOwnProperty('Task Owner')) {
        obj['Task Owner'] = ApiClient.convertToType(data['Task Owner'], 'String');
      }
    }
    return obj;
  }


  /**
   * @member {String} Accounts
   */
  exports.prototype['Accounts'] = undefined;

  /**
   * @member {String} Contacts/Leads
   */
  exports.prototype['Contacts/Leads'] = undefined;

  /**
   * @member {String} Description
   */
  exports.prototype['Description'] = undefined;

  /**
   * @member {Date} Due Date
   */
  exports.prototype['Due Date'] = undefined;

  /**
   * @member {String} Priority
   */
  exports.prototype['Priority'] = undefined;

  /**
   * @member {Boolean} Recurring Activity
   */
  exports.prototype['Recurring Activity'] = undefined;

  /**
   * @member {Boolean} Remind At
   */
  exports.prototype['Remind At'] = undefined;

  /**
   * @member {Boolean} Send Notification Email
   */
  exports.prototype['Send Notification Email'] = undefined;

  /**
   * @member {String} Status
   */
  exports.prototype['Status'] = undefined;

  /**
   * @member {String} Subject
   */
  exports.prototype['Subject'] = undefined;

  /**
   * @member {String} Task Owner
   */
  exports.prototype['Task Owner'] = undefined;




  return exports;
}));
